### PR TITLE
cron task abort [WIP]

### LINKED ISSUE #1910: cron timeout task should not free execution-plan

### PR DESCRIPTION
**Pull request by swilly22**
resolves #1910

### CODE REVIEW

File: src/util/cron.c
Comment (jeffreylovitz) [inline src/util/cron.c:229]: I don't understand the purpose of this cast - why do `CRON_TASK` and `CronTaskHandle` need to be interchangable?

File: src/util/cron.c
Comment (jeffreylovitz) [inline src/util/cron.c:131]: What a useful function!

**Review by jeffreylovitz — APPROVED** [review]
:+1: I really like the design here, and all the local testing I've performed indicates that it works as intended.

I wouldn't mind if we could take some time to go over this online, as there are some areas I don't follow perfectly. That can take place before or after merge, however.

File: src/util/cron.c
Comment (AviAvni) [inline src/util/cron.c:237]: lock all queries for long time

File: src/util/cron.c
Comment (swilly22) [reply src/util/cron.c:229]: I don't want to revel `CRON_TASK` outside of this file, this type is internal

File: src/util/cron.c
Comment (swilly22) [reply src/util/cron.c:237]: I don't think so, (given the current tasks we're issuing)
In addition it would hold this lock for a "long" period of time, when the caller of this function is aborting when the task is executing, the chances of that are not high.

File: src/util/cron.c
Comment (swilly22) [reply src/util/cron.c:247]: Because if we break, caller might be messing with the task's private data
e.g. freeing the execution-plan being drained.

**Pull request merged**
